test/Common/Test.hs

```diff
@@ -2441,3 +2446,27 @@ shouldBeOnClauseWithoutMatchingJoinException ea =
             pure ()
         _ ->
             expectationFailure $ "Expected OnClauseWithMatchingJoinException, got: " <> show ea
+
+testOverloadedRecordDot :: SpecDb
+testOverloadedRecordDot = describe "OverloadedRecordDot" $ do
+#if __GLASGOW_HASKELL__ >= 902
+    describe "with SqlExpr (Entity rec)" $ do
+        itDb "lets you project from a record" $ do
+            select $ do
+                bp <- Experimental.from $ table @BlogPost
+                pure bp.title
+    describe "with SqlExpr (Maybe (Entity rec))" $ do
+        itDb "lets you project from a Maybe record" $ do
+            select $ do
+                p :& mbp <- Experimental.from $
+                    table @Person
+                    `leftJoin` table @BlogPost
+                        `Experimental.on` do
+                            \(p :& mbp) ->
+                                just p.id ==. mbp.authorId
+                pure (p.id, mbp.title)
```

Comment:
im gonna pop some `tags`
only got twenty dollars in my pocket
i'm i'm i'm hunting
lookin for a come up
this is fucking awesome~